sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala

```diff
@@ -319,6 +319,19 @@ class AstBuilder extends DataTypeAstBuilder
     RepeatStatement(condition, body, Some(labelText))
   }
 
+  override def visitForStatement(ctx: ForStatementContext): ForStatement = {
```

Comment:
nit: Keep in mind Danko's PR for label checks, because it will probably merge before this, so just add the checks here as well.